src/overview/components/DateRangeSelection.jsx

```diff
@@ -1,46 +1,125 @@
-import React, { PropTypes } from 'react'
+import React, { Component, PropTypes } from 'react'
 import moment from 'moment'
+import chrono from 'chrono-node'
 import DatePicker from 'react-datepicker'
 import 'react-datepicker/dist/react-datepicker-cssmodules.css'
 
 import styles from './DateRangeSelection.css'
 
 
-const DateRangeSelection = ({
-    startDate, endDate,
-    onStartDateChange, onEndDateChange,
-}) => (
-    <div className={styles.dateRangeSelection}>
-        <DatePicker
-            className={styles.datePicker}
-            dateFormat='DD-MM-YYYY'
-            placeholderText='after..'
-            isClearable
-            selected={startDate && moment(startDate)}
-            selectsStart
-            startDate={moment(startDate || 0)}
-            endDate={moment(endDate)}
-            maxDate={moment()}
-            onChange={date => onStartDateChange(
-                date ? date.valueOf() : undefined,
-            )}
-        />
-        <DatePicker
-            className={styles.datePicker}
-            dateFormat='DD-MM-YYYY'
-            placeholderText='before..'
-            isClearable
-            selected={endDate && moment(endDate)}
-            selectsEnd
-            startDate={moment(startDate || 0)}
-            endDate={moment(endDate)}
-            maxDate={moment()}
-            onChange={date => onEndDateChange(
-                date ? date.endOf('day').valueOf() : undefined,
-            )}
-        />
-    </div>
-)
+class DateRangeSelection extends Component {
+    constructor() {
+        super()
+
+        this.state = {
+            startDateText: null,
+            endDateText: null,
+        }
+
+        this.handleKeydown = this.handleKeydown.bind(this)
+        this.submitDateChange = this.submitDateChange.bind(this)
+    }
+
+    componentDidMount() {
+        this.refs.startDateDatePicker.refs.input.refs.input.addEventListener('keydown', e => this.handleKeydown(e, 'startDate'))
+        this.refs.endDateDatePicker.refs.input.refs.input.addEventListener('keydown', e => this.handleKeydown(e, 'endDate'))
+    }
+
+    handleKeydown(event, type) {
+        if (event.key === 'Enter') {
+            event.stopImmediatePropagation()
+            this.submitDateChange(type)
+        }
+    }
+
+    submitDateChange(type) {
```

Comment:
This method seems a good candidate for DRYing out, as it's just the same code repeated apart from `state.startDateText`, `props.startDate`, `props.onStartDateChange` (and their `endDate` counterparts). Might as well extract them out, switching on `type` in the method body, and return a function that runs the main logic in here (bound to those vars), if that makes sense? If this method returns a function, the calls don't need to be nested in an arrow function (L90/117)